vortex-serde/src/layouts/read/layouts.rs

```diff
@@ -189,26 +189,31 @@ impl LayoutReader for ColumnLayout {
                     .children()
                     .ok_or_else(|| vortex_err!("Missing children"))?;
 
-                let column_layouts = match self.scan.projection {
-                    Projection::All => (0..fb_children.len())
-                        .map(|idx| self.read_child(idx, fb_children, s.dtypes()[idx].clone()))
-                        .collect::<VortexResult<Vec<_>>>()?,
-                    Projection::Flat(ref v) => v
-                        .iter()
-                        .zip(s.dtypes().iter().cloned())
-                        .map(|(projected_field, dtype)| {
-                            let child_idx = match projected_field {
-                                Field::Name(n) => s.find_name(n.as_ref()).ok_or_else(|| {
-                                    vortex_err!("Invalid projection, trying to select  {n}")
-                                })?,
-                                Field::Index(i) => *i,
-                            };
-                            self.read_child(child_idx, fb_children, dtype)
-                        })
-                        .collect::<VortexResult<Vec<_>>>()?,
+                let (names, column_layouts) = match &self.scan.projection {
+                    Projection::All => {
+                        let layouts = (0..fb_children.len())
+                            .zip_eq(struct_dtype.dtypes().iter())
+                            .map(|(index, dtype)| {
+                                self.read_child(index, fb_children, dtype.clone())
+                            })
+                            .collect::<VortexResult<Vec<_>>>()?;
+                        (struct_dtype.names().clone(), layouts)
+                    }
+                    Projection::Flat(projected_fields) => {
+                        let mut names = Vec::with_capacity(projected_fields.len());
+                        let mut layouts = Vec::with_capacity(projected_fields.len());
+                        for projected_field in projected_fields {
+                            let FieldInfo { index, name, dtype } =
+                                struct_dtype.field_info(projected_field)?;
```

Comment:
yes this, I had a change that just projected dtype and the re resolved fields here